docs/assembly.rst

```diff
@@ -391,12 +394,35 @@ will have a wrong impression about the stack height at label ``two``:
 .. code::
 
     {
+        let x := 8
         jump(two)
         one:
-            // Here the stack height is 1 (because we pushed 7),
-            // but the assembler thinks it is 0 because it reads
+            // Here the stack height is 2 (because we pushed x and 7),
+            // but the assembler thinks it is 1 because it reads
             // from top to bottom.
-            // Accessing stack variables here will lead to errors.
+            // Accessing the stack variable x here will lead to errors.
+            x := 9
+            jump(three)
+        two:
+            7 // push something onto the stack
+            jump(one)
+        three:
+    }
```

Comment:
Practically, I would suggest, instead of introducing labels with height adjustments, to seek another way of translation where less syntactic entities appear.

1. transform the program so that every new stack element has its own block.
2. introduce jumps and labels only when their block-depths match

The example here would look like
```
{
   let x := 8
   jump(two)
   { 
     one:
     x := 9
   }

   two:
   { 7
     jump(one)
   }

   three:
}
```